turtleGraphics.js

```diff
@@ -0,0 +1,244 @@
+
+    //Changing the direction to the right of the current direction
+    right() {
+        switch (this.direction) {
+            case 'east':
+                this.direction = 'south';
+                break;
+            case 'south': 
+                this.direction = 'west';
+                break;
+            case 'west':
+                this.direction = 'north';
+                break;
+            case 'north': 
+                this.direction = 'east';
+                break;
+        }
+        return this;
+    }
+
+    //Changing the direction to the left of the current direction
+    left() {
+        switch (this.direction) {
+            case 'east':
+                this.direction = 'north';
+                break;
+            case 'south': 
+                this.direction = 'east';
+                break;
+            case 'west':
+                this.direction = 'south';
+                break;
+            case 'north': 
+                this.direction = 'west';
+                break;
+        }
+        return this;
+    }
+
+    //Printing out the steps the turtle took to make the path it made
+    allPoints() {
+        console.log(this.array);
+    }
+
+    //Printing out the steps the turtle took on a grid that starts at (0,0) and goes to the max height and width that the turtle reached
+    print() {
+        let maxNorth = 0;
+        let maxSouth = 0;
+        let maxEast = 0;
+        let maxWest = 0;
+        //String of the path to be printed
+        let string = '';
+        //Boolean to check whether the turtle has already taken a step or not
+        let walk = false;
+        //Array of coordinates to be printed
+        let arrayToPrint = this.array;
+        //Getting the maximum width and height (negative and posititve) that the turtle walks
+        for (let i in this.array) {
+            if (maxEast < this.array[i][0]) {
+                maxEast = this.array[i][0];
+            }
+            if (maxWest > this.array[i][0]) {
+                maxWest = this.array[i][0];
+            }
+            if (maxSouth < this.array[i][1]){
+                maxSouth = this.array[i][1];
+            }
+            if (maxNorth > this.array[i][1]) {
+                maxNorth = this.array[i][1];
+            }
+        }
+        //Getting rid of duplicate steps in the path for printing
+        for (let i in arrayToPrint) {
+            for (let j in arrayToPrint) {
+                if (i !== j && arrayToPrint[i][0] === arrayToPrint[j][0] && arrayToPrint[i][1] === arrayToPrint[j][1]) {
+                    // if (j == arrayToPrint.length-1) {
+                    //     arrayToPrint.splice(i, 1);
+                    // } else {
+                        arrayToPrint.splice(j, 1);
+                    //}
+                }
+            }
+        }
+        //Iterating through the min and max width and heights to print out the grid and the turtle steps
+        for (let i = maxNorth; i <= maxSouth; i++) {
```

Comment:
avoid using nested loops. it's fine in this exercise as the sample is small but in real practice it degrades performance.